File: forensics-btmgen/src/test/java/de/burger/forensics/plugin/GlobRegexCacheTest.java
// DEST: src/test/java/de/burger/forensics/plugin/GlobRegexCacheTest.java
package de.burger.forensics.plugin;

import static de.burger.forensics.plugin.GlobUtils.globToRegexCached;
import static org.junit.jupiter.api.Assertions.assertNotSame;
import static org.junit.jupiter.api.Assertions.assertSame;

import org.junit.jupiter.api.Test;

class GlobRegexCacheTest {

    @Test
    void returnsIdenticalInstanceForIdenticalGlob() {
        Object first = globToRegexCached("**/*.java");
        Object second = globToRegexCached("**/*.java");

        assertSame(first, second, "Cache should return the same instance for identical glob");
    }

    @Test
    void returnsDifferentInstancesForDifferentGlobs() {
        Object javaGlob = globToRegexCached("**/*.java");
        Object kotlinGlob = globToRegexCached("**/*.kt");

        assertNotSame(javaGlob, kotlinGlob, "Different globs should yield different cached objects");
    }
}

File: forensics-btmgen/src/test/kotlin/de/burger/forensics/plugin/GlobRegexCacheTest.kt
package de.burger.forensics.plugin

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class GlobRegexCacheTest {
    @Test
    fun `returns identical instance for identical glob`() {
        val first = globToRegexCached("**/*.java")
        val second = globToRegexCached("**/*.java")

        assertThat(first).isSameAs(second)
    }

    @Test
    fun `returns different instances for different globs`() {
        val javaGlob = globToRegexCached("**/*.java")
        val kotlinGlob = globToRegexCached("**/*.kt")

        assertThat(javaGlob).isNotSameAs(kotlinGlob)
    }
}
